db/config.cc

```diff
@@ -674,6 +674,12 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , enable_shard_aware_drivers(this, "enable_shard_aware_drivers", value_status::Used, true, "Enable native transport drivers to use connection-per-shard for better performance")
     , enable_ipv6_dns_lookup(this, "enable_ipv6_lookup", value_status::Used, false, "Use IPv6 address resolution")
     , abort_on_internal_error(this, "abort_on_internal_error", liveness::LiveUpdate, value_status::Used, false, "Abort the server instead of throwing exception when internal invariants are violated")
+    , max_partition_keys_per_query(this, "max_partition_keys_per_query", liveness::LiveUpdate, value_status::Used, 100,
+            "Maximum number of distinct partition keys per query. This limit places a bound on the size of IN tuples, "
+            "especially when multiple partition key columns have IN restrictions. Increasing this value can result in server instability.")
+    , max_clustering_keys_per_query(this, "max_clustering_keys_per_query", liveness::LiveUpdate, value_status::Used, 100,
```

Comment:
This name is a bit misleading it may create an impression that it will limit the number of keys returned by a query with no restrictions. I know there's a description but maybe a better name would help avoid users misunderstand the option.

Reply:
Yes, I thought of that too but did not find a better name.

Reply:
@glommer perhaps you have a suggestion for a better name?

Reply:
why not `max_partition_restrictions_per_in_query` ?

Reply:
I don't like it (nobody will guess what restrictions are, it's an internal term) but it's better that what we have no so I'll change to it.